### PR TITLE
Standardized UnitClass definition in TranslationStore subclasses

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

File: translate/storage/csvl10n.py [hunk 1]
```diff
@@ -352,8 +352,7 @@ class csvfile(base.TranslationStore):
     Extensions = ["csv"]
 
     def __init__(self, inputfile=None, fieldnames=None, encoding=None):
-        base.TranslationStore.__init__(self, unitclass=self.UnitClass)
-        self.units = []
+        super(csvfile, self).__init__()
         self.encoding = encoding or 'auto'
```
Comment (unho) [inline translate/storage/csvl10n.py]: Why not passing the encoding to `__init__()`?

File: translate/storage/subtitles.py [hunk 2]
```diff
@@ -79,11 +79,9 @@ class SubtitleFile(base.TranslationStore):
     """A subtitle file"""
     UnitClass = SubtitleUnit
 
-    def __init__(self, inputfile=None, unitclass=UnitClass):
+    def __init__(self, inputfile=None, **kwargs):
         """construct an Subtitle file, optionally reading in from inputfile."""
-        self.UnitClass = unitclass
-        base.TranslationStore.__init__(self, unitclass=unitclass)
-        self.units = []
+        super(SubtitleFile, self).__init__(self, **kwargs)
```
Comment (unho) [inline translate/storage/subtitles.py]: Why `__init__(self, **kwargs)` and not just `__init__(**kwargs)`?

**Comment by unho** [comment]
Nice cleanup. Will merge after all doubts/nits are addressed and PR is rebased.

**Comment by claudep** [comment]
Fixed what you noticed and rebased.

**Comment by unho** [comment]
I am going to merge this despite that perhaps the `unitclass` argument perhaps can also be removed from `base.TranslationStore.__init__()`. But that can be done in a future PR if considered a good idea.

**Comment by unho** [comment]
Merged. Thanks!

**Comment by claudep** [comment]
I don't know either if `unitclass` as a constructor argument is useful or even used. However, it is currently part of the API, so if any third-party is currently calling SomeStoreClass(unitclass=customUnitClass), we would break compatibility if we remove that possibility.

**Comment by unho** [comment]
Sure.

**Pull request merged**